rpc/auth.go

```diff
@@ -200,6 +200,60 @@ func MakeSimpleMultiAuthHandler(forEntities, expectedPayloads []string) AuthHand
 	})
 }
 
+// MakeSimpleMultiAuthPairHandler works similarly to MakeSimpleMultiAuthHandler with the addition of
+// supporting a key, id pair used to ensure that a key that maps to the id matches the key passed
+// during the function call.
+func MakeSimpleMultiAuthPairHandler(forEntities, expectedPayloads []map[string]string) AuthHandler {
```

Comment:
(sorry left a comment here originally and then deleted it while i was looking at the pr a bit more but re-adding something in the similar vein) 

can you explain a bit about how you are planning on calling this? for example, what is an expected values for the `forEntities` map and the `expectedPayloads` map? seems like what we would want to accept multiple keyid-key pairs (where the keyid is the entity and the key is the payload) in only one map?  

once i have clarity on this I feel like itll help me comment more on the rest

Reply:
forEntities would contain a bucket of (keyID, key) mappings that we are basically comparing the `expectedPayloads` against to 1) ensure that the expectedPayload(keyID, key mapping) exists in bucket 2) ensure that the keyID found has a key that matches the key passed into the expectedPayload. The payload is a keyId and key pair and our goal here is to ensure that that keyID exists in a bucket of keyIDs that we're searching through and when we find that keyID, we want to verify that the key passed in the payload is the same as the one stored in the bucket. Not sure if this helps, more than happy to discuss in person.

Reply:
so expectedEntities being a map of keyID to key makes sense to me. but expectedPayloads will be passed in when this auth handler is initialized right? so we are comparing the arguments of `AuthHandlerFunc` to `forEntities` and `expectedPayloads`? 

basically, what i'm getting at, is why do there have to be two maps of keyId to key? in `AuthHandlerFunc` the entity should be a keyId and `payload` should be the key itself, so seems like one map would suffice here?

Reply:
also can discuss monday in person as it may be easier to explain then